XRTK-Core/Packages/com.xrtk.core/Runtime/Providers/Controllers/Hands/HandTrackedPoseProcessor.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) XRTK. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using XRTK.Definitions.Controllers.Hands;
+using XRTK.Definitions.Utilities;
+using XRTK.Extensions;
+
+namespace XRTK.Providers.Controllers.Hands
+{
+    /// <summary>
+    /// The hand pose processor uses the recorded hand pose definitions
+    /// configured in <see cref="Definitions.InputSystem.MixedRealityInputSystemProfile.TrackedPoses"/>
+    /// or the platform's <see cref="BaseHandControllerDataProviderProfile.TrackedPoses"/>
+    /// and attempts to recognize a hand's current pose during runtime to provide for
+    /// <see cref="HandData.TrackedPose"/>.
+    /// </summary>
+    public sealed class HandTrackedPoseProcessor
+    {
+        /// <summary>
+        /// Creates a new recognizer instance to work on the provided list of poses.
+        /// </summary>
+        /// <param name="recognizablePoses">Recognizable poses by this recognizer.</param>
+        public HandTrackedPoseProcessor(IReadOnlyList<HandControllerPoseDefinition> recognizablePoses)
+        {
+            bakedHandDatas = new HandData[recognizablePoses.Count];
+            definitions = new Dictionary<int, HandControllerPoseDefinition>();
+
+            var i = 0;
+            foreach (var item in recognizablePoses)
```

Comment:
Can we use a for loop here so we avoid any additional IEnumerable allocs?

Reply:
We can I guess. Bus this runs only in the constructor once.

Reply:
Done, thanks